artemis-commons/src/main/java/org/apache/activemq/artemis/utils/critical/CriticalMeasure.java

```diff
@@ -17,36 +17,48 @@
 
 package org.apache.activemq.artemis.utils.critical;
 
-import org.jboss.logging.Logger;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 
 public class CriticalMeasure {
 
-   private static final Logger logger = Logger.getLogger(CriticalMeasure.class);
+   //uses updaters to avoid creates many AtomicLong instances
+   private static final AtomicLongFieldUpdater<CriticalMeasure> TIME_ENTER_UPDATER = AtomicLongFieldUpdater.newUpdater(CriticalMeasure.class, "timeEnter");
+   private static final AtomicLongFieldUpdater<CriticalMeasure> TIME_LEFT_UPDATER = AtomicLongFieldUpdater.newUpdater(CriticalMeasure.class, "timeLeft");
 
    private volatile long timeEnter;
```

Comment:
I thought you said you were going to use MIN_VALUE here?

Reply:
I've pushed a PR about this (sad) fact I've discovered reading the nanoTime javadoc: this is the correct way to do this...

Reply:
this one: https://github.com/apache/activemq-artemis/pull/1518
I wasn't aware about this behaviour of System::nanoTime: having received a good review has helped me to dig into it and find it :+1:

Reply:
timeEnter shouldn't be defaulted to MInvalue?  I didn't see any updates on that PR

Reply:
In that PR are fixed other parts that rely on the wrong mechanics of nanoTime, while on this one it is taken in account here: 
https://github.com/apache/activemq-artemis/pull/1515/files/a2189e0f6b74a40c1975fe2c0e15605d87b1c55e#diff-8a85ed9c24cc23c3b4230cae837e4797R34

With nanoTime there is no MIN_VALUE (absolute I mean) sadly: that's the point of that PR.
ATM I don't have much time to explain it better, so please check the javadoc I've linked there...

Reply:
All I'm referring is the fact you said you would set timeEnter and timeLeft as Long.MIN_VALUE (There's certainly a possible MIN_VALUE in long):
```java
   private volatile long timeEnter = Long.MIN_VALUE;
   private volatile long timeLeft = Long.MIN_VALUE;```

Reply:
I just realized what you did.. you set enterCritical and leaveCritical on constructor...

I asked you a simple question... why there's no MIN_VALUE as you said you would do... then  you started answering something else.. and tell me you don't have time??? ;)